ntbs-service/Services/AdImportService.cs

```diff
@@ -29,11 +30,23 @@ public async Task RunCaseManagerImportAsync()
             var tbServices = await _referenceDataRepository.GetAllTbServicesAsync();
             using (var adDirectoryService = _adDirectoryServiceFactory.Create())
             {
-                foreach (var (user, tbServicesMatchingGroups) in adDirectoryService.LookupUsers(tbServices))
+                var users = adDirectoryService.LookupUsers(tbServices).ToList();
+                foreach (var (user, tbServicesMatchingGroups) in users)
                 {
                     Log.Information($"Updating user {user.Username}");
                     await _userRepository.AddOrUpdateUser(user, tbServicesMatchingGroups);
                 }
+                
+                var ntbsUsersNotInAd = this._userRepository.GetUserIQueryable()
+                    .Where(user => !users.Select(u => u.user.Username)
+                        .Contains(user.Username)).ToList();
+                foreach (var user in ntbsUsersNotInAd)
+                {
+                    Log.Information($"Updating user {user.Username}");
+                    user.IsActive = false;
+                    user.AdGroups = null;
+                    await _userRepository.AddOrUpdateUser(user, user.CaseManagerTbServices.Select(cmtb => cmtb.TbService));
+                }

```

Comment:
Another minor thing, but I think it would be more natural to have a single `foreach` loop, with an `if-else` statement inside, rather than two `foreach` loops.

I find it a bit weird that we're setting some users to an invalid state and then overriding them. (I'm not sure if AddOrUpdateUser is actually saving the changes in the database, but if it is then I find this even more troubling... as it would be possible - though very unlikely - for the invalid state to actually be fetched).

Reply:
Which invalid state are you referring to? AddOrUpdateUser is saving changes.